modules/languages/ruby.nix

```diff
@@ -0,0 +1,157 @@
+{
+  pkgs,
+  config,
+  lib,
+  ...
+}:
+with lib;
+with builtins; let
+  cfg = config.vim.languages.ruby;
+
+  defaultServer = "rubyserver";
+  servers = {
+    rubyserver = {
+      package = pkgs.rubyPackages_3_2.solargraph;
+      lspConfig = ''
+        lspconfig.rubyserver.setup {
+          capabilities = capabilities;
+          on_attach = attach_keymaps,
+          cmd = { "${pkgs.bundler}/bin/bundle exec solargraph", "stdio" }
+        }
+      '';
+    };
+  };
+
+  # TODO: specify packages
+  defaultFormat = "rubocop";
+  formats = {
+    rubocop = {
+      package = pkgs.rubocop;
+      nullConfig = ''
+
+        local conditional = function(fn)
+          local utils = require("null-ls.utils").make_conditional_utils()
+          return fn(utils)
+        end
+
+        table.insert(
+          ls_sources,
+                     null_ls.builtins.formatting.rubocop.with({
+                         command = "${pkgs.bundler}/bin/bundle",
+                      args = vim.list_extend(
+                          { "exec", "rubocop", "-a" },
+                          null_ls.builtins.formatting.rubocop._opts.args
+                        ),
+                     })
+
+
+          -- conditional(function(utils)
+          --   return utils.root_has_file("Gemfile")
+          --           and null_ls.builtins.formatting.rubocop.with({
+          --               command = "${pkgs.bundler}",
+           --            args = vim.list_extend(
+          --                { "exec", "rubocop" },
+          --                nls.builtins.formatting.rubocop._opts.args
+          --              ),
+          --           })
+          --           or null_ls.builtins.formatting.rubocop.with({
+          --             command = "${cfg.format.package}/bin/rubocop",
+         --          })
+          -- end)
+        )
+      '';
+    };
+  };
+  # TODO: specify packages
+  defaultDiagnostics = ["rubocop"];
+  diagnostics = {
+    rubocop = {
+      package = pkgs.rubyPackages_3_2.rubocop;
```

Comment:
but here it is `pkgs.rubyPackages_3_2.rubocop` - is there a specific reason for this, or can we match both packages to avoid pulling extra inputs?

Reply:
Diagnostic is going away when this all works